validator/client/beacon-api/submit_aggregate_selection_proof.go

```diff
@@ -0,0 +1,80 @@
+package beacon_api
+
+import (
+	"context"
+	"net/url"
+	"strconv"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/prysm/v3/beacon-chain/core/helpers"
+	"github.com/prysmaticlabs/prysm/v3/beacon-chain/rpc/apimiddleware"
+	"github.com/prysmaticlabs/prysm/v3/consensus-types/primitives"
+	ethpb "github.com/prysmaticlabs/prysm/v3/proto/prysm/v1alpha1"
+	"github.com/prysmaticlabs/prysm/v3/time/slots"
+)
+
+func (c *beaconApiValidatorClient) submitAggregateSelectionProof(ctx context.Context, in *ethpb.AggregateSelectionRequest) (*ethpb.AggregateSelectionResponse, error) {
+	validatorIndexResponse, err := c.validatorIndex(ctx, &ethpb.ValidatorIndexRequest{PublicKey: in.PublicKey})
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get validator index")
+	}
+
+	attesterDuties, err := c.dutiesProvider.GetAttesterDuties(ctx, slots.ToEpoch(in.Slot), []primitives.ValidatorIndex{validatorIndexResponse.Index})
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get attester duties")
+	}
+
+	if len(attesterDuties) == 0 {
+		return nil, errors.Errorf("no attester duty for the given slot %d", in.Slot)
+	}
+
+	// First attester duty is required since we requested attester duties for one validator index.
+	attesterDuty := attesterDuties[0]
+
+	committeeLen, err := strconv.ParseUint(attesterDuty.CommitteeLength, 10, 64)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to parse committee length")
+	}
+
+	isAggregator, err := helpers.IsAggregator(committeeLen, in.SlotSignature)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get aggregator status")
+	}
+	if !isAggregator {
+		return nil, errors.New("validator is not an aggregator")
+	}
```

Comment:
I don't think we need this part. The API endpoint should be responsible for validating such conditions.

Reply:
yeah, I initially had the same thought but when I checked its rpc counterpart then this condition was there so included it here too. Also beacon APIs don't have the functionality of validating this condition since we don't post slot signature in any endpoint: https://github.com/prysmaticlabs/prysm/blob/71b4468be6dcc53fb4ef5481ccecfae04da8b8d1/beacon-chain/rpc/prysm/v1alpha1/validator/aggregator.go#L61

Reply:
Looking at the description of https://ethereum.github.io/beacon-APIs/?urls.primaryName=dev#/Validator/getAggregatedAttestation, there is no constraint as to who can fetch the data. If you think this is incorrect, you should open an issue here: https://github.com/ethereum/beacon-APIs

One thing that we are missing in https://github.com/prysmaticlabs/prysm/blob/63186c8b0f67ca30641a1d8315882c1e8dfe2115/beacon-chain/rpc/eth/validator/validator.go#L741 is this part:
> A 503 error must be returned if the block identified by the response beacon_block_root is optimistic (i.e. the aggregated attestation attests to a block that has not been fully verified by an execution engine).

It would be nice to fix this either in this PR or another one.

Reply:
No, that [getAggregatedAttestation](https://ethereum.github.io/beacon-APIs/?urls.primaryName=dev#/Validator/getAggregatedAttestation) endpoint is correct. In beacon API compatible flow, validator [checks](https://github.com/ethereum/consensus-specs/blob/dev/specs/phase0/validator.md#aggregation-selection) if it is an aggregator or not and if it is an aggregator then it subscribes to committee subnet with this endpoint: https://ethereum.github.io/beacon-APIs/?urls.primaryName=dev#/Validator/prepareBeaconCommitteeSubnet by passing `is_aggregator` to true.

In prysm, the `is_aggregator` check happens to be on the beacon node side when prysm validator calls `SubmitAggregateSelectionProof` which is why I included this in my PR. Ideally this check should be on validator side and if it is NOT an aggregator then it should do nothing. Please correct me if I am getting something wrong in my interpretation.

> It would be nice to fix this either in this PR or another one.

Sure

Reply:
Done!